src/devices/src/virtio/block/simple_handler.rs

```diff
@@ -0,0 +1,99 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR BSD-3-Clause
+
+use std::result;
+
+use log::warn;
+use vm_memory::{self, Bytes, GuestAddressSpace};
+use vm_virtio::{DescriptorChain, Queue};
+
+use crate::virtio::block::request::{DiskProperties, Request};
+use crate::virtio::SignalUsedQueue;
+
+#[derive(Debug)]
+pub enum Error {
+    GuestMemory(vm_memory::GuestMemoryError),
+    Queue(vm_virtio::Error),
+}
+
+impl From<vm_memory::GuestMemoryError> for Error {
+    fn from(e: vm_memory::GuestMemoryError) -> Self {
+        Error::GuestMemory(e)
+    }
+}
+
+impl From<vm_virtio::Error> for Error {
+    fn from(e: vm_virtio::Error) -> Self {
+        Error::Queue(e)
+    }
+}
+
+// This object is used to process the queue of a block device without making any assumptions
+// about the notification mechanism. We're using the backend from Firecracker for now (the
+// `DiskProperties` object), but the aim is to have a way of working with generic backends
+// to turn this into a more useful building block.
+pub struct SimpleHandler<M: GuestAddressSpace, S: SignalUsedQueue> {
+    pub driver_notify: S,
+    pub queue: Queue<M>,
+    pub disk: DiskProperties,
+}
+
+impl<M: GuestAddressSpace, S: SignalUsedQueue> SimpleHandler<M, S> {
+    fn process_chain(&mut self, mut chain: DescriptorChain<M>) -> result::Result<(), Error> {
+        match Request::parse(&mut chain) {
+            Ok(request) => {
+                let len;
+                // TODO: The executor could actually write the status itself, right?
```

Comment:
It should, still trying to figure out how to do this in a nice way :D.

Reply:
That's cool; we can def live with this as is for the time being.